contracts/eosio.bios/src/eosio.bios.cpp

```diff
@@ -54,4 +56,124 @@ void bios::reqactivated( const eosio::checksum256& feature_digest ) {
+   resource_config_singleton.set(config, get_self());
+}
+
+void bios::buyram(eosio::name dao_owner, eosio::name app, eosio::asset quant) {
+    require_auth(app); // Check that the app has the necessary authorization
+
+   // Access the account table from id.tmy.hpp
+    idtmy::id::account_type_table account_type(idtmy_name, idtmy_name.value);
+   // Check the account type of the app
+   auto itr = account_type.find(app.value);
+   eosio::check(itr != account_type.end(), "Could not find account");
+   eosio::check(itr->account_type == idtmy::enum_account_type::App, "Only apps can buy and sell RAM");
+
+   // Check that the RAM is being purchased with the correct token
+   eosio::check(quant.symbol == bios::system_resource_currency, "must buy ram with core token");
+
+   // Check that the amount of tokens being used for the purchase is positive
+   eosio::check(quant.amount > 0, "Amount must be positive");
+    // Check the decimal place of the token amount
+    eosio::check(quant.symbol.precision() != 4, "Token amount must have up to 4 decimal places");
+
+    // Get the RAM price
+   resource_config_table resource_config_singleton(get_self(), get_self().value);
+   eosio::check(resource_config_singleton.exists(), "Resource config must be initialized before use");
+    auto config = resource_config_singleton.get();
+
+    // Read values from the table
+    double ram_price = config.ram_price;
+    uint64_t ram_fee = (1.0 + config.ram_fee);
+    uint64_t ram_purchase = (ram_price / ram_fee) * quant.amount;
+
+    eosio::check(config.total_ram_available >= config.total_ram_used + ram_purchase, "Not enough RAM available");
+
+    // modify the values and save them back to the table,
+    config.total_ram_used += ram_purchase;
+    resource_config_singleton.set(config, get_self());
+    
+    //Allocate the RAM
+    int64_t myRAM, myNET, myCPU;
+    eosio::get_resource_limits( app, myRAM, myNET, myCPU );
+    eosio::set_resource_limits(app, myRAM + ram_purchase, myNET, myNET);
+
+
+   // Transfer token and buy ram
+     eosio::action(permission_level{get_self(), "active"_n},
+               "onocoin.tmy"_n,
+               "transfer"_n,
+        std::make_tuple(dao_owner, gov_name, quant, std::string("buy ram")))
+            .send();
+}
+
+void bios::sellram(eosio::name dao_owner, eosio::name app, eosio::asset quant) {
+    require_auth(app); // Check that the app has the necessary authorization
+
+    // Access the account table from id.tmy.hpp
+    idtmy::id::account_type_table account_type("id.tmy"_n, "id.tmy"_n.value);
+    // Check the account type of the app
+    auto itr = account_type.find(app.value);
+    eosio::check(itr != account_type.end(), "Could not find account");
+    eosio::check(itr->account_type == idtmy::enum_account_type::App, "Only apps can buy and sell RAM");
+
+    // Check that the amount of bytes being sold is positive
+    eosio::check(quant.amount > 0, "Amount must be positive");
+
```

Comment:
you are missing `   eosio::check(quant.symbol == bios::system_resource_currency, "must buy ram with core token");` like you have in `buyram()` please check you have the same checks as relevant